### DISCUSSION TITLE
Sytemrequirements / MariaDB Database

### DISCUSSION THREAD

**Discussion by alexh-swdev** [OpenMage/magento-lts#3962] [Questions & Answers]
Hi,

we are currently running MariaDB 10.3(.39)

Can I savely upgrade to 10.11 or 11.0 (or any other 11.x) or are there any known issues with specific versions? (Just asking, because other competitors report problems e.g. with  10.11.5 und 11.0.3)

Alex

**Comment by Flyingmana** [comment]
do you have links for these reports?
Sure is worth we check them what might apply to us just to be sure

**Reply by alexh-swdev** [reply]
Just to make sure: by "other competitors" I meant other shop systems ;)
Specifically Shopware mentioned issues: https://developer.shopware.com/docs/guides/installation/requirements.html linking to https://jira.mariadb.org/browse/MDEV-31931

**Reply by Flyingmana** [reply]
> Just to make sure: by "other competitors" I meant other shop systems ;)

I thought so already 🙂
its ok here to link stuff of other shop systems for reference, usually there is always something to learn 

> Specifically Shopware mentioned issues: https://developer.shopware.com/docs/guides/installation/requirements.html linking to https://jira.mariadb.org/browse/MDEV-31931

this seems to be specific to generated columns? need to look more into what got reported there, but I think we are not using this feature yet anyway.

**Reply by alexh-swdev** [reply]
I don't now if this should better be a bug report, but as we are talking about the db here: From time to time, when bringing my test system in sync with the live system, I stumple upon an invalid option. For ```example catalog_product_relation``` has the create option ```ROW_FORMAT=FIXED``` which seems not to be supported since a while (during import it throws errno 140 "Wrong create options" )
Can this get removed during an update by some kind of migration? (I don't know much about the OM database/resource scripts) I also don't know whether there is a proper equivalent?

**Reply by fballiano** [reply]
it doesn't seem that FIXED is not available anymore:
https://dev.mysql.com/doc/refman/8.0/en/static-format.html

but

> Fixed-length row format is available only for tables having no BLOB or TEXT columns. Creating a table having such columns with an explicit ROW_FORMAT clause does not raise an error or warning; the format specification is ignored.

does your table have TEXT/BLOB columns created by some module?

**Reply by alexh-swdev** [reply]
No. Those look like core tables to me:
````
$installer->getTable('catalog_product_website')} in mysql4-upgrade-0.7.36-0.7.37.php
$installer->getTable('catalog_product_website')} in mysql4-upgrade-1.4.0.0.4-1.4.0.0.5.php

$installer->getTable('catalog/product_relation')} in mysql4-upgrade-1.4.0.0.4-1.4.0.0.5.php
````

I figured out, FIXED is a MyIsam thing and those tables are InnoDB. It looks like InnoDB does not support that at all: https://mariadb.com/kb/en/innodb-row-formats/
and perhaps, here MySQL and MariaDB differ in the error handling (MySQL might just issue a warning, idk, but Maria throws an error).

**Reply by fballiano** [reply]
ah ok that's a difference between maria and mysql because I've never seen an error on mysql.

so either we convert that table to myisam or to another row format?

**Reply by alexh-swdev** [reply]
The tables have foreign keys. If I remember correctly, MyISAM does (did?) not support those.

Here (some Mac-SQL-App) is Magento mentioned, too:
https://github.com/sequelpro/sequelpro/issues/2478#issuecomment-211290812
Those people say, FIXED, appart from not beeing support, does not make sense for InnoDB tables.
I am no database export, but reading this, I would remove it (or ask you to remove it :) because I think this must be done via such a migration and I don't know how those work in O.M. )

**Reply by Flyingmana** [reply]
the reason to use fixed in the past was related to how table lookups worked, and that tables were stored on classic harddisks.
Latest with the new nvme storage types there is likely no notable benefit anymore.(purely amateur point of view from me)

I would also say we can go forward without the fixed row type.
